src/ast/codegen_llvm.cpp

```diff
@@ -1373,26 +1380,32 @@ void CodegenLLVM::visit(FieldAccess &acc)
     return;
   }
 
-  std::string cast_type = type.is_tparg ? tracepoint_struct_ : type.cast_type;
+  std::string cast_type = is_tparg ? tracepoint_struct_ : type.GetName();
   Struct &cstruct = bpftrace_.structs_[cast_type];
 
-  type = SizedType(type);
-  type.size = cstruct.size;
-  type.cast_type = cast_type;
+  // This overwrites the stored type!
+  type = CreateRecord(cstruct.size, cast_type);
+  if (is_ctx)
+    type.MarkCtxAccess();
+  type.is_tparg = is_tparg;
+  type.is_internal = is_internal;
+  type.is_kfarg = is_kfarg;
```

Comment:
> I realize it might be out of the scope of this commit, but couldn't all this be done in the semantic analyser?

It should be. I think this is the result of probe expansion happening in all phases instead of early on. 

> Also, why are the variables copied and reassigned? Doesn't look like changes are being made to them.

`type` is a references so the `CreateRecord()` call overwrites the attributes. Which causes issues with the wildcard expansion